### PR TITLE
Add horreum clients reference documentation

### PR DESCRIPTION
**Pull request by lampajr**
## Fixes Issue

Fixes https://github.com/Hyperfoil/Horreum/issues/1551

## Changes proposed

This PR aims to add an initial version of Horreum client documentation under `References` section.

## Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by lampajr** [comment]
This is how it would look like:
image
image

**Comment by lampajr** [comment]
Hey @stalep , IIRC you were mentioning about this addition in Horreum doc. This is a first draft, hope that's what you were looking for :crossed_fingers:

**Comment by stalep** [comment]
> Hey @stalep , IIRC you were mentioning about this addition in Horreum doc. This is a first draft, hope that's what you were looking for 🤞

It looks very good!!

File: docs/site/content/en/docs/Reference/horreum-clients.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+// async call
+v, err := client.RawClient.Api().Config().Version().Get(context.Background(), nil)
+if err != nil {
+    log.Fatalf("error getting server version: %s", err.Error())
+    os.Exit(1)
+}
+```
+
+Similarly to the Pythong example, the previous async call was the equivalent of doing
```
[the remark this answers is not in the document]
Comment (lampajr) [reply docs/site/content/en/docs/Reference/horreum-clients.md:110]: thx, good catch!

**Pull request merged**